src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Uap.cs

```diff
@@ -15,4 +18,50 @@ internal static partial class ProcessManager
     {
         public static IntPtr GetMainWindowHandle(int processId) => IntPtr.Zero;
     }
+
+    internal static partial class NtProcessManager
+    {
+        private static ProcessModuleCollection GetModules(int processId, bool firstModuleOnly)
+        {
+            // We don't have a good way of getting all of the modules of the particular process,
+            // but we can at least get the path to the executable file for the process, and
+            // other than for debugging tools, that's the main reason consumers of Modules care about it,
+            // and why MainModule exists.
+            try
+            {
+                ArrayPool<char> pool = ArrayPool<char>.Create(maxArraysPerBucket: 1, maxArrayLength: 64);
+                char[] chars;
+                int capacity = 64, length = 0;
+
+                // Get the path to the executable
+                using (Process process = Process.GetProcessById(processId))
+                {
+                    do
+                    {
+                        capacity *= 2;
+                        chars = pool.Rent(capacity);
+                        length = Interop.Kernel32.GetModuleFileNameEx(process.SafeHandle, IntPtr.Zero, chars, capacity);
```

Comment:
The buffer returned by the pool can be bigger than what you have asked for. You should pass the actual size of the buffer to the PInvoke (ie `chars.Length` instead of `capacity`).